app/models/manageiq/providers/vmware/infra_manager/provision_workflow.rb

```diff
@@ -144,6 +144,16 @@ def available_vlans_and_hosts(options = {})
     return vlans, hosts
   end
 
+  def filter_hosts_by_vlan_name(all_hosts)
+    vlan_name = get_value(@values[:vlan])
+    return all_hosts unless vlan_name
+
+    _log.info("Filtering hosts with the following network: <#{vlan_name}>")
+    shared = !vlan_name.match(/dvs_/).nil?
+    vlan_name.sub!(/^dvs_/, '') if shared
+    all_hosts.select { |h| h.lans.any? { |lan| lan.name == vlan_name && !lan.switch.shared.nil? == shared } }
```

Comment:
Well now this counts on `switch.shared` being nil and will break if we update the parser in the future to (I think) correctly set shared to false.
`!!lan.switch.shared` will treat nil and false as "falsy" which is what we want.